templates/docs/examples/patterns/side-navigation/_expand_script.js

```diff
@@ -0,0 +1,43 @@
+// Setup expand toggles for all side navigations on the page.
+
+var expandToggles = document.querySelectorAll('.p-side-navigation__expand');
+var navigationLinks = document.querySelectorAll('.p-side-navigation__link');
+
+const setup = (toggle) => {
+  if (!toggle.hasAttribute('aria-expanded')) {
+    toggle.setAttribute('aria-expanded', false);
+  }
+  const item = toggle.closest('.p-side-navigation__item');
+  const link = item.querySelector('.p-side-navigation__link');
+  const nestedList = item.querySelector('.p-side-navigation__list');
+  if (!link?.hasAttribute('aria-expanded')) {
+    link.setAttribute('aria-expanded', false);
+  }
+  if (!nestedList?.hasAttribute('aria-expanded')) {
+    nestedList.setAttribute('aria-expanded', false);
+  }
+};
+
+const handleToggle = (e) => {
+  const item = e.currentTarget.closest('.p-side-navigation__item');
+  const button = item.querySelector('.p-side-navigation__expand');
+  const link = item.querySelector('.p-side-navigation__link');
+  const nestedList = item.querySelector('.p-side-navigation__list');
```

Comment:
Is `querySelector` going to always return first closest element in the tree? The lists can be nested even more, so technically any item may have multiple `.p-side-navigation__list` nested in it. We want the one closest to the item.

Is `querySelector` going to return that by definition?

Reply:
Good point. Even though technically `querySelector` doesn't guarantee to return the first closest element in the tree (as it's using depth-first traversal) this isn't a problem in this case.

That's because by definition - any nested `p-side-navigation__list` will always be contained in outer list which will be returned first as the browser traverses down the tree.

Reply:
Yes, that's also my understanding, so let's hope it works like that in all implementations :)